### PR TITLE
Async announcements caching for attachments-lacking posts 

### LINKED ISSUE #5: Missed Attachments in post nr. 490
Spotted a bug regarding document attachments in the bot. 
During the run of the 10:20 26/04/2022 the bot correctly posted the announcement with number 490 but was unable to send the related attachments.

### PR DESCRIPTION
**Pull request by LukeTheWalker**
This PR should provide a small announcement caching logic to prevent what has been observed in #5.

Whenever a post is lacking their related attachments it is inserted inside a txt file called `cached_announcements.txt` and in each subsequent run of the bot it will be checked to see if any attachment was uploaded even if it is no more inside the scope of the traditional search. 

Also added union typing to chat_id argument.

Closes #5

### CODE REVIEW

File: albo.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ def send_telegram_attachments(pdf_links: List[str]) -> None:
 # Special Headers in which is preferable to put a break line character to separate section of tg message
 break_line_headers = ["Oggetto", "Inizio pubblicazione"]
 
-for id in range (last_id + 1, new_id + 1):
+for id in cached_announcements + list(range(last_id + 1, new_id + 1)):
```
Comment (Helias) [inline albo.py:80]: this is gonna be very big, move in a separate variable please

File: albo.py [hunk 2]
```diff
@@ -97,5 +104,7 @@ def send_telegram_attachments(pdf_links: List[str]) -> None:
       send_telegram_message(error_string, dev)
   #print(message)
 
-with open("last_id.txt", "w+") as f:
-    f.write(str(new_id))
+with open("last_id.txt", "w+") as f_id, open("cached_announcements.txt", "w+") as f_cached:
+    f_id.write(str(new_id))
+    for cached_id in cached_announcements:
+      f_cached.write(str(cached_id) +  "\n")
```
Comment (Helias) [inline albo.py:110]: from your point of view, does it has sense to move the second file inside the first `with open()`?
```python
with open("last_id.txt", "w+") as f_id:
    f_id.write(str(new_id))
   with open("cached_announcements.txt", "w+") as f_cached:
        for cached_id in cached_announcements:
          f_cached.write(str(cached_id) +  "\n")
```

Comment (Helias) [reply albo.py:110]: or.. make it separate?

```python
with open("last_id.txt", "w+") as f_id:
    f_id.write(str(new_id))

with open("cached_announcements.txt", "w+") as f_cached:
    for cached_id in cached_announcements:
        f_cached.write(str(cached_id) +  "\n")
```

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
is this ready?

**Comment by LukeTheWalker** [comment]
> is this ready?

It is ready but I was able to test it only on synthetically created inputs, I was waiting for the albo to publish an announcement without attachments to test it on some real world data but since I implemented this PR it never occurred. If we feel confident enough about it we could merge it even now as-is.

**Comment by Helias** [comment]
I suggest to put in ignore all the txt files

**Review by Helias — APPROVED** [review]

**Pull request merged**